substrate/pkg/utils/kubectl/apply.go

```diff
@@ -0,0 +1,61 @@
+package kubectl
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *Client) Apply(ctx context.Context, file string) error {
+	resp, err := http.Get(file)
+	if err != nil {
+		return fmt.Errorf("getting response, %w", err)
+	}
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("parsing response, %w", err)
+	}
+	if err := c.ApplyYAML(ctx, data); err != nil {
+		return fmt.Errorf("applying yaml, %w", err)
+	}
+	logging.FromContext(ctx).Infof("Applied %s", file)
+	return nil
+}
+
+func (c *Client) ApplyYAML(ctx context.Context, yamlBytes []byte) error {
+	objects, err := c.ParseObjects(ctx, yamlBytes)
+	if err != nil {
+		return fmt.Errorf("parsing objects, %w", err)
+	}
+	for _, object := range objects {
+		if err := c.DeleteObject(ctx, object); err != nil {
```

Comment:
Not clear why are we deleting the objects first and then creating them again. Does this mean every time user runs this cli it will delete and recreate all these objects?

Reply:
Let's discuss this one offline. I only had a couple of bad solutions to this one.